Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/EndToEndTraceWriterBaselineTests.TypedPointOperationsAsync.xml

```
@@ -109,7 +110,15 @@
                           "duration in milliseconds": 0,
                           "data": {
                             "Client Side Request Stats": "Redacted To Not Change The Baselines From Run To Run"
-                          }
+                          },
+                          "children": [
```

It's odd that this is in a `children` list of its own?

---

That's the way it's designed to keep it consistent.

---

Looking at this again this seems like wrong child trace, because it doesn't deserialize until after the pipeline not all the way at the bottom. This is likely caused by the use the ResponseMessage.Trace which was passed in at that low level.

---

It might but those diagnostics are already broken and this change would actually fix those bugs. Creating child on a disposed parent should throw an exception, but that might break some scenarios.